src/sound/YMF278.hh

```diff
@@ -134,12 +144,17 @@ private:
 
 	// To speed-up memory access we divide the 4MB address-space in 32 blocks of 128kB.
 	// For each block we point to the corresponding region inside 'rom' or 'ram',
-	// or nullptr means this region is unmapped.
+	// or 'nullBlock' means this region is unmapped.
 	// Note: we can only _read_ via these pointers, not write.
-	std::array<const uint8_t*, 32> memPtrs = {};
+	std::array<Block128, 32> memPtrs{
+		nullBlock, nullBlock, nullBlock, nullBlock, nullBlock, nullBlock, nullBlock, nullBlock,
+		nullBlock, nullBlock, nullBlock, nullBlock, nullBlock, nullBlock, nullBlock, nullBlock,
+		nullBlock, nullBlock, nullBlock, nullBlock, nullBlock, nullBlock, nullBlock, nullBlock,
+		nullBlock, nullBlock, nullBlock, nullBlock, nullBlock, nullBlock, nullBlock, nullBlock,
+	};
```

Comment:
One would expect that there is a shorter way to express this...

Reply:
The technical problem here is that Block128 has no default constructor. So we cannot first declare the array and then e.g. fill it with the correct value.

I know there are some c++ tricks using `std::index_sequence` and parameter packs. But my first idea was that those would add more complexity than just these 4 lines with 32 repeats of `nullBlock`.

However, I checked and in our `stl.hh` we _do_ already have a helper function `generate_array` that implements exactly this trick. With that helper, I think, we could simplify this code snippet to (not tested):
```
auto memPtrs = generate_array<32>([](size_t){ return nullBlock; });
```

I'll experiment and change it.